File: CodeWars_Ruby_solution/6kyu/Who likes it？.rb
def likes(names)
    ans = String.new
    case names.length
      when 0
      ans = "no one likes this"
      when 1
      ans = names[0] + " likes this"
      when 2
      ans = names[0] + " and " + names[1] + " like this"
      when 3
      ans = names[0] + ", " + names[1] + " and " + names[2] + " like this"
      else
      ans = names[0] + ", " + names[1] + " and #{names.length - 2} others like this"
    end
    ans
  end

File: CodeWars_Cpp_solution/6 kyu/Who likes it？.cpp
std::string likes(const std::vector<std::string> &names)
{
  std::string ans = "";
  int len = names.size();
  switch (len) {
      case 0:
        ans = "no one likes this";
        break;
      case 1:
        ans = names[0] + " likes this";
        break;
      case 2:
        ans = names[0] + " and " + names[1] + " like this";
        break;
      case 3:
        ans = names[0] + ", " + names[1] + " and " + names[2] + " like this";
        break;
      default:
        ans = names[0] + ", " + names[1] + " and "+ std::to_string(len - 2) + " others like this";
        break;
  }
  return ans; // Do your magic!
}